standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java

```diff
@@ -1259,6 +1261,43 @@ public boolean dropTable(String catName, String dbName, String tableName)
     return success;
   }
 
+  private void markTableAsPendingDrop(String catName, String dbName, String tableName) {
```

Comment:
Do you actually need to mark table as dropped or you could simply remove the record?